lib/file_management.js

```diff
@@ -0,0 +1,124 @@
+const archethic = require('archethic')
+const chalk = require('chalk')
+const crypto = require('crypto')
+const fs = require('fs')
+const path = require('path')
+const jsdom = require("jsdom")
+const {
+    JSDOM
+} = jsdom
+const mime = require('mime')
+
+function confirmationCallback(endpoint, address, message, file) {
+    return (nbConfirmations) => {
+        if (nbConfirmations == 1) {
+            console.log(chalk.green(message))
+            console.log(chalk.blue(endpoint + "/api/last_transaction/" + address + "/content?mime=" + mime.getType(file)))
+        }
+        console.log(chalk.magenta("Transaction confirmed with " + nbConfirmations + " replications"))
+    }
+
+}
+
+
+module.exports = {
+
+    hmac: function (Files, algo, myseed, Seed, Address) {
+        for (let i = 0; i < Files.length; i++) {
+            const hmac = crypto.createHmac(algo, myseed);
+            hmac.update(Files[i])
+            const seed = hmac.digest('hex')
+            Seed.push(seed)
+            const address = archethic.deriveAddress(seed, 0)
+            Address.push(address)
+        }
+    },
+
+    hmac2: function (x, algo, myseed) {
+
+        const hmac = crypto.createHmac(algo, myseed);
+        hmac.update(x)
+        const seed = hmac.digest('hex')
+        const address = archethic.deriveAddress(seed, 0)
+
+        return {
+            seed,
+            address
+        }
+
+    },
+
+
+    readdir: function (Directory, Files) {
+        try {
+            fs.readdirSync(Directory).forEach(File => {
+                const abs = path.join(Directory, File);
+                if (fs.statSync(abs).isDirectory()) return ReadDirectory(abs);
+                else return Files.push(abs);
+            });
+        } catch (e) {
+            console.error(chalk.red(e.message))
+            return
+        }
+    },
+
+    convert_file_to_transaction: async (folder, array_files, array_address, endpoint) => {
```

Comment:
Here the behavior is not to convert file to transaction, but to replace links with transaction address references.
So you can see, the naming can bring confusion.